packages/legend-studio/src/stores/sidebar-state/LocalChangesState.ts

```diff
@@ -244,6 +322,102 @@ export class LocalChangesState {
     }
   }
 
+  *refreshWorkspaceSyncStatus(): GeneratorFn<void> {
+    try {
+      yield flowResult(
+        this.sdlcState.fetchRemoteWorkspaceRevision(
+          this.sdlcState.activeProject.projectId,
+          this.sdlcState.activeWorkspace,
+        ),
+      );
+      if (this.sdlcState.isWorkspaceOutOfSync) {
+        const remoteWorkspaceEntities =
+          (yield this.editorStore.sdlcServerClient.getEntitiesByRevision(
+            this.sdlcState.activeProject.projectId,
+            this.sdlcState.activeWorkspace,
+            this.sdlcState.activeRemoteWorkspaceRevision.id,
+          )) as Entity[];
+        this.editorStore.changeDetectionState.workspaceRemoteLatestRevisionState.setEntities(
+          remoteWorkspaceEntities,
+        );
+        yield flowResult(
+          this.editorStore.changeDetectionState.workspaceRemoteLatestRevisionState.buildEntityHashesIndex(
+            remoteWorkspaceEntities,
+            LogEvent.create(
+              CHANGE_DETECTION_LOG_EVENT.CHANGE_DETECTION_LOCAL_HASHES_INDEX_BUILT,
+            ),
+          ),
+        );
+        yield flowResult(
+          this.editorStore.changeDetectionState.computeAggregatedWorkspaceRemoteChanges(),
+        );
+      } else {
+        this.editorStore.changeDetectionState.workspaceRemoteLatestRevisionState.setEntities(
+          [],
+        );
+        this.editorStore.changeDetectionState.setpotentialWorkspaceSyncConflicts(
+          [],
+        );
+        this.editorStore.changeDetectionState.setAggregatedWorkspaceRemoteChanges(
+          [],
+        );
+      }
+    } catch (error) {
+      assertErrorThrown(error);
+      this.editorStore.applicationStore.log.error(
+        LogEvent.create(LEGEND_STUDIO_LOG_EVENT_TYPE.SDLC_MANAGER_FAILURE),
+        error,
+      );
+    }
+  }
+
+  openPotentialWorkspaceSyncConflict(conflict: EntityChangeConflict): void {
```

Comment:
is this function get used anywhere? especially when we don't even show the potential conflicts anywhere.

Reply:
Yes. This is used in the preview you wanted to add so users can see incoming remote changes on the sidebar.